pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java

```diff
@@ -142,6 +142,14 @@ public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceE
                 }
             };
 
+    private static final FastThreadLocal<Map<Consumer, Set<Integer>>> localConsumerStickyKeyHashesMap =
```

Comment:
why do we need this ThreadLocal ?
here we are recycling a simple HashMap

we are "saving" a "new HashMap" and in order to do it we are adding more code and memory footprint.
The JVM should be able to handle this HashMap efficiently, probably within the TLAB 

do you have evidence that this threadlocal helps ?

Reply:
No, I don't have clear evidence. I just implemented it the same as the existing `localGroupedEntries`.
https://github.com/apache/pulsar/blob/40c831761a7519e4ce9f0b7b8808cdb6df3f418e/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentStickyKeyDispatcherMultipleConsumers.java#L143-L149

`localGroupedEntries` seems to have been added in the following PR:
https://github.com/apache/pulsar/pull/7104

@merlimat, do we need to use ThreadLocal in this case?

Reply:
@merlimat ping

Reply:
Fixed not to use FastThreadLocal.